core/src/main/java/hudson/model/Queue.java

```diff
@@ -2461,8 +2461,8 @@ public int compareTo(WaitingItem that) {
         }
 
         public CauseOfBlockage getCauseOfBlockage() {
-            long diff = timestamp.getTimeInMillis() - System.currentTimeMillis();
-            if (diff > 0)
+            long diff = System.currentTimeMillis() - timestamp.getTimeInMillis();
+            if (diff <= 10000)
```

Comment:
10 seconds seems excessive. There's clearly something wrong if this occurs. I think at most 100 millis grace period make sense, and then I would fake a "0" label rather than go with a negative number.